### PR TITLE
SpectralConv2DTranspose

### PR DESCRIPTION
**Pull request by cofri**
The `SpectralConv2DTranspose` layer is a new Lipschitz layer equivalent to the tf.keras `Conv2DTranspose`. 
Note that some arguments must be set to a specific value to be supported:
- `padding` must be `same`
- `dilation_rate` must be `(1, 1)`
- `output_padding` must be `None`

Like the SpectralConv2D layer, the kernel is normalized using the spectral normalization, then orthogonalized with Björck algorithm.

### CODE REVIEW

File: tests/test_layers.py [hunk 1]
```diff
@@ -187,7 +189,7 @@ def train_k_lip_model(
     np.random.seed(42)
     # create the keras model, defin opt, and compile it
     model = generate_k_lip_model(layer_type, layer_params, input_shape, k_lip_model)
-    optimizer = Adam(lr=0.001)
+    optimizer = Adam(learning_rate=0.001)
```
Comment (thib-s) [inline tests/test_layers.py:192]: maybe `Adam(0.001)` would do the job when changing tensorflow version

**Review by thib-s — COMMENTED** [review]
LGTM excepted from leaked code from branch `ci_multiple_tf_versions`. Might be good to remove it (especially as tests for this PR seems to be ran using old config)

File: tests/test_layers.py (hunk 1 above)
Comment (cofri) [reply tests/test_layers.py:192]: I've updated this to avoid warning of deprecated argument, After discussions, as the arg name `learning_rate` exists since TensorFlow 2.0, it should be ok for all supported TF versions in deel-lip.

**Review by thib-s — APPROVED** [review]

**Comment by thib-s** [comment]
rebased branch onto develop
